File: parser/props/DNABattleCell.py
class DNABattleCell:
    PROP_CODE = 21

    def __init__(self, width, height, pos):
        self.width = width
        self.height = height
        self.pos = pos

    def __str__(self):
        return 'DNABattleCell width: {0}, height: {1}, pos: {2}'.format(
            self.getWidth(), self.getHeight(), self.getPos())

    def setWidth(self, width):
        self.width = width

    def getWidth(self):
        return self.width

    def setHeight(self, height):
        self.height = height

    def getHeight(self):
        return self.height

    def setPos(self, pos):
        self.pos = pos

    def getPos(self):
        return self.pos

    def setWidthHeight(self, width, height):
        self.setWidth(width)
        self.setHeight(height)

File: reader/DNABattleCell.cpp
#include "DNABattleCell.h"

DNABattleCell::DNABattleCell() {};
DNABattleCell::~DNABattleCell() {};

void DNABattleCell::set_width(unsigned char width)
{
	m_width = width;
};

void DNABattleCell::set_height(unsigned char height)
{
	DNABattleCell::m_height = height;
};

void DNABattleCell::set_width_height(unsigned char width, unsigned char height)
{
	set_width(width);
	set_height(height);
};

void DNABattleCell::set_pos(LVector3f* pos)
{
	m_pos = pos;
};

unsigned char DNABattleCell::get_width()
{
	return m_width;
};

unsigned char DNABattleCell::get_height()
{
	return m_height;
};

LVector3f* DNABattleCell::get_pos()
{
	return m_pos;
};
